src/Workspaces/Core/Portable/Workspace/Solution/SourceGeneratedDocumentState.cs

```diff
@@ -7,120 +7,136 @@
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis
+namespace Microsoft.CodeAnalysis;
+
+internal sealed class SourceGeneratedDocumentState : DocumentState
 {
-    internal sealed class SourceGeneratedDocumentState : DocumentState
+    private readonly Lazy<Checksum> _lazyTextChecksum;
+
+    public SourceGeneratedDocumentIdentity Identity { get; }
+
+    public string HintName => Identity.HintName;
+
+    /// <summary>
+    /// It's reasonable to capture 'text' here and keep it alive.  We're already holding onto the generated text
+    /// strongly in the ConstantTextAndVersionSource we're passing to our base type. 
+    /// </summary>
+    public SourceText SourceText { get; }
```

Comment:
this simplified code which was previously trying to be resilient to the source-text not being there (which can't ever happen.